Zend/zend_exceptions.c

```diff
@@ -175,6 +175,12 @@ ZEND_API ZEND_COLD void zend_throw_exception_internal(zend_object *exception) /*
 
 	if (exception != NULL) {
 		zend_object *previous = EG(exception);
+		if (previous && (zend_is_unwind_exit(previous) || zend_is_graceful_exit(previous))) {
```

Comment:
Maybe this should be limited to unwind_exit only, and leave graceful_exit alone?

Reply:
So, I may treat unwind_exit() as SIGKILL and graceful_exit() as SIGTERM, it seems reasonable for me in a way... graceful_exit() could  be aborted due to other accidents.

Reply:
@nikic This is limited to internal/extension code, yes? User code would not be able to throw another exception and replace the exception while destroying a fiber?

Reply:
User code can throw an exception in finally. Though that would work independently of this code.

Reply:
Seems I had a [test the user code behavior](https://github.com/php/php-src/blob/86a0d4b22d471c38b4ce1a80b92611b5e0950dd8/Zend/tests/fibers/unfinished-fiber-with-throw-in-finally.phpt#L1). Obviously this didn't affect that and since this only concerns throwing an exception internally, never mind about my question of storing the exception.

@nikic I like your suggestion of only preventing replacement of unwind_exit.